File: Swift/problem005.swift
// Problem 5 - Project Euler
// http://projecteuler.net/index.php?section=problems&id=5
func gcd(_ x: Int, _ y: Int) -> Int {
    return y == 0 ? x : gcd(y, x % y)
}

func lcm(_ x: Int, _ y: Int) -> Int {
    return (x == 0 || y == 0) ? 0 : x * y / gcd(x, y)
}

print((1...20).reduce(1, lcm))

File: Elixir/problem005.exs
# Problem 5 - Project Euler
# http://projecteuler.net/index.php?section=problems&id=5
defmodule P005 do
  def gcd(a, 0), do: a
  def gcd(0, b), do: b
  def gcd(a, b), do: gcd(b, rem(a, b))

  def lcm(0, 0), do: 0
  def lcm(a, b), do: div(a * b, gcd(a, b))

  def solve() do
    1..20
    |> Enum.reduce(10, &lcm/2)
  end
end

IO.puts P005.solve()
